raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -215,6 +227,22 @@ public void maybeUpdateElectionLatency(long currentTimeMs) {
         }
     }
 
+    public <T> void addLeaderMetrics(LeaderState<T> leaderState) {
+        metrics.addMetric(numObserversMetricName, (Gauge<Integer>) (config, now) -> leaderState.numObservers());
+        metrics.addMetric(uncommittedVoterChangeMetricName, (Gauge<Integer>) (config, now) -> {
+            if (leaderState.addVoterHandlerState().isEmpty() && leaderState.removeVoterHandlerState().isEmpty()) {
+                return 0;
+            } else {
+                return 1;
+            }
+        });
+    }
```

Comment:
This lambda are not thread safe. `LeaderState` assumes that the kraft driver thread is the only reader and writer.